ITE/4/README.adoc

```diff
@@ -430,35 +432,80 @@ resulting build and CI report.
 ==== Use Case 3: Granular JSON Access
 
 The enhancement proposed in this ITE can also be used to provide more granular
-access to certain resources, such as JSON files. A generic URI can be used to
-resolve to the information contained in a specific key of a JSON file. This allows
-for greater flexibility when using in-toto with different types of artifacts.
+access to resources such as JSON files. A generic URI can be used to refer to
+information contained in a specific key of a JSON file; in this scenario, each
+field serves as an abstract resource.
+
+A practical example of this use case is for Cloud Native Application Bundles
+(CNAB) metadata. A link:https://github.com/cnabio/cnab-spec/blob/master/101-bundle-json.md[CNAB bundle.json file]
+contains information such as the top-level package information, images included
+in the bundle, credentials required by the application, and so on. We define a
+minimal supply chain described below for the creation of these bundle.json
+files. This supply chain is used in CNAB Signy and can be
+link:https://github.com/cnabio/signy/blob/master/scripts/in-toto/minimal.py[verified using in-toto].
+A link:https://github.com/adityasaky/in-toto/tree/ite-4-cnab-resolver[prototype resolver]
+was created using the structure defined in the <<uri-resolvers, URI Resolvers>>
+section for CNAB bundle.json files.
+
+===== in-toto link attestation for the creation of the initial bundle.json
+
+In this step of the minimal supply chain, a developer creates an initial
+bundle.json file with some details such as the images to use populated. However,
+the specific properties of the images are to be filled in by the machine as part
+of the next step.
 
-===== in-toto link attestation signing contents of a specific JSON key
+```
+{
+    "_type": "link",
+    "byproducts": {},
+    "command": [],
+    "environment": {},
+    "materials": {},
+    "name": "developer",
+    "products": {
+        "...": "...",
+        "cnab+json:/home/saky/Work/secure-systems-lab/cnab-ite-4/bundle.json$/images/my-microservice/contentDigest": <HASH>,
```

Comment:
I disagree that the CNAB use case can be covered by this TAP. A URI is supposed to address a document, and not necessarily parts of it, unless we refer to anchor parts (which has a standard notation). This looks like "abusing" the URI scheme to address elements inside a JSON document, but perhaps I am interpreting the meaning of the URI scheme too strictly.

Reply:
Very technically, RFC 3986 does not apply any such limits.

```
Resource

      This specification does not limit the scope of what might be a
      resource; rather, the term "resource" is used in a general sense
      for whatever might be identified by a URI.  Familiar examples
      include an electronic document, an image, a source of information
      with a consistent purpose (e.g., "today's weather report for Los
      Angeles"), a service (e.g., an HTTP-to-SMS gateway), and a
      collection of other resources.  A resource is not necessarily
      accessible via the Internet; e.g., human beings, corporations, and
      bound books in a library can also be resources.  Likewise,
      abstract concepts can be resources, such as the operators and
      operands of a mathematical equation, the types of a relationship
      (e.g., "parent" or "employee"), or numeric values (e.g., zero,
      one, and infinity).
```

I believe that the CNAB use case must be covered by ITE-4 and such a use case was certainly intended in spirit when it was written. It may, however, be worth looking again at the proposed URI format for the CNAB use case again, as well as expanding on the general format (which just talks about `scheme` and `hier-part`) as a whole.

Reply:
To expand on the intended in spirit bit, I think some of the same questions can be raised for artifacts in SPDX documents as well, which was also considered when writing this ITE.